test/e2e/helloworld_test.go

```diff
@@ -104,3 +106,119 @@ func TestHelloWorld(t *testing.T) {
 		t.Fatalf("Failed to get Service name from Revision label")
 	}
 }
+
+func TestHelloWorldUserPort(t *testing.T) {
+	var userPort int32
+	userPort = 8888
+	clients := Setup(t)
+
+	//add test case specific name to its own logger
+	logger := logging.GetContextLogger("TestHelloWorld-UserPort")
+
+	var imagePath = test.ImagePath("helloworld-userport")
+
+	logger.Infof("Creating a new Route and Configuration")
+	names, err := CreateRouteAndConfig(clients, logger, imagePath, &test.Options{
+		ContainerPorts: generateUserPort(logger, userPort),
```

Comment:
Since you're already creating a new YAML file, would it be better to move this setting to the YAML file instead?

Reply:
hmm, It is another way to write this test. ok, I will refer to helloworld_shell_test.go to modify this test. Thank you for your every comment.